liquibase-couchbase/src/test/java/system/change/MutateInQuerySystemTest.java

```diff
@@ -0,0 +1,90 @@
+import static common.constants.TestConstants.TEST_COLLECTION_3;
+import static common.constants.TestConstants.TEST_SCOPE;
+import static common.matchers.CouchbaseCollectionAssert.assertThat;
+import static common.operators.TestCollectionOperator.createOneFieldJson;
+import static liquibase.ext.couchbase.types.Document.document;
+
+public class MutateInQuerySystemTest extends LiquibaseSystemTest {
+
+    private static TestCollectionOperator testCollectionOperator;
+
+    private static final JsonObject expectedReplaceContent = createOneFieldJson("newDocumentField", "newDocumentValue");
+    private static final String[] documentIds = {"replaceDocument1", "replaceDocument2", "replaceDocument3"};
+
+    @BeforeEach
+    void cleanUpd() {
+        for (String docId : documentIds) {
+            if (testCollectionOperator.docExists(docId)) {
+                testCollectionOperator.getCollection().remove(docId);
+            }
+        }
+    }
+
+    @AfterAll
+    static void cleanIndex() {
+        DropPrimaryQueryIndexOptions options = dropPrimaryQueryIndexOptions().ignoreIfNotExists(true);
+        testCollectionOperator.dropPrimaryIndex(options);
+    }
+
+    @BeforeAll
+    static void createPrimaryIndex() throws InterruptedException {
+        bucketOperator.createCollection(TEST_COLLECTION_3, TEST_SCOPE);
+        testCollectionOperator = bucketOperator.getCollectionOperator(TEST_COLLECTION_3, TEST_SCOPE);
+        TimeUnit.SECONDS.sleep(2L);
+        testCollectionOperator.createPrimaryIndex();
+        TimeUnit.SECONDS.sleep(2L);
+    }
+
+    @Test
+    @SneakyThrows
+    void Should_replace_document() {
+        Document doc = document(documentIds[0], createOneFieldJson("aKey", "avalue"));
+        testCollectionOperator.insertDoc(doc);
+
+        Liquibase liquibase = liquibase(MUTATE_QUERY_FILTER_IN_REPLACE_DOCUMENT_TEST_XML);
+        liquibase.update();
+
+        Document expected = document(doc.getId(), expectedReplaceContent);
+        assertThat(testCollectionOperator.getCollection()).contains(expected);
+    }
+
+    @Test
+    @SneakyThrows
+    void Should_replace_documents() {
+        Document doc1 = document(documentIds[0], createOneFieldJson("aKey1", "avalue"));
+        Document doc2 = document(documentIds[1], createOneFieldJson("aKey1", "avalue"));
+        Document doc3 = document(documentIds[2], createOneFieldJson("aKey1", "avalue5"));
+        testCollectionOperator.insertDocs(doc1, doc2, doc3);
+
+        Liquibase liquibase = liquibase(MUTATE_QUERY_FILTER_IN_REPLACE_DOCUMENTS_TEST_XML);
+        liquibase.update();
+
+        Document expected1 = document(doc1.getId(), expectedReplaceContent);
+        assertThat(testCollectionOperator.getCollection()).contains(expected1);
+
+        Document expected2 = document(doc2.getId(), expectedReplaceContent);
+        assertThat(testCollectionOperator.getCollection()).contains(expected2);
+
+        assertThat(testCollectionOperator.getCollection()).contains(doc3);
```

Comment:
Small proposal

I propose to move expected documents to prepare part (at the beginning of the test) 
```
  Document doc1 = document(documentIds[0], createOneFieldJson("aKey1", "avalue"));
  Document doc2 = document(documentIds[1], createOneFieldJson("aKey1", "avalue"));
  Document doc3 = document(documentIds[2], createOneFieldJson("aKey1", "avalue5"));
  testCollectionOperator.insertDocs(doc1, doc2, doc3);

  Document expected1 = document(doc1.getId(), expectedReplaceContent);
  Document expected2 = document(doc2.getId(), expectedReplaceContent);

  Liquibase liquibase = liquibase(MUTATE_QUERY_FILTER_IN_REPLACE_DOCUMENTS_TEST_XML);
  liquibase.update();

  assertThat(testCollectionOperator.getCollection()).contains(asList(expected1, expected2, doc3));
```